tests/cpptests/micro-benchmarks/benchmark_doc_id_pattern_iteration.cpp

```diff
@@ -0,0 +1,241 @@
+                } else if (idDistributionType == SPARSE_JUMPS_100) {
+                    unionData[unionIdx][idListIdx][docIdx] = baseId + (docIdx + 1) * 100;
+                }
+            }
+            std::sort(unionData[unionIdx][idListIdx].begin(), unionData[unionIdx][idListIdx].end());
+        }
+    }
+
+public:
+    // Helper function to create union iterators with idlist children
+    IndexIterator* createUnionIterator(size_t unionIdx) {
+        const size_t numIdLists = unionData[unionIdx].size();
+        IndexIterator **idListIterators = createIdListIterators(unionIdx, numIdLists);
+        return NewUnionIterator(idListIterators, numIdLists, true, 1.0, QN_UNION, NULL, &RSGlobalConfig.iteratorsConfigParams);
+    }
+
+    // Helper function to create intersection iterator with two union children
+    IndexIterator* createIntersectionIterator() {
+        IndexIterator **unionIterators = (IndexIterator **)rm_malloc(sizeof(IndexIterator *) * 2);
+        unionIterators[0] = createUnionIterator(0);
+        unionIterators[1] = createUnionIterator(1);
+        return NewIntersectIterator(unionIterators, 2, NULL, RS_FIELDMASK_ALL, -1, 0, 1.0);
+    }
+
+private:
+    // Create array of idlist iterators for a union
+    IndexIterator** createIdListIterators(size_t unionIdx, size_t numIdLists) {
+        IndexIterator **idListIterators = (IndexIterator **)rm_malloc(sizeof(IndexIterator *) * numIdLists);
+
+        for (size_t i = 0; i < numIdLists; i++) {
+            t_docId *ids = copyIds(unionData[unionIdx][i]);
+            idListIterators[i] = NewIdListIterator(ids, unionData[unionIdx][i].size(), 1.0);
+        }
+
+        return idListIterators;
+    }
+
+    // Create a copy of IDs for an idlist iterator
+    t_docId* copyIds(const std::vector<t_docId>& sourceIds) {
+        t_docId *ids = (t_docId *)rm_malloc(sizeof(t_docId) * sourceIds.size());
+        for (size_t i = 0; i < sourceIds.size(); i++) {
+            ids[i] = sourceIds[i];
+        }
+        return ids;
+    }
+};
+
+bool BM_IntersectionIterator::initialized = false;
+
+// Benchmark scenarios:
+// Parameter 0: Number of idlist iterators per union (10, 25, 50)
+// Parameter 1: Number of documents per idlist iterator (1000, 5000)
+// Parameter 2: ID distribution type (CONSECUTIVE, SPARSE_JUMPS_100, CONSECUTIVE_MODULO)
+#define INTERSECTION_SCENARIOS() \
+    Args({10, 1000, CONSECUTIVE})->Args({10, 1000, SPARSE_JUMPS_100})->Args({10, 1000, CONSECUTIVE_MODULO})-> \
+    Args({10, 5000, CONSECUTIVE})->Args({10, 5000, SPARSE_JUMPS_100})->Args({10, 5000, CONSECUTIVE_MODULO})-> \
+    Args({25, 1000, CONSECUTIVE})->Args({25, 1000, SPARSE_JUMPS_100})->Args({25, 1000, CONSECUTIVE_MODULO})-> \
+    Args({25, 5000, CONSECUTIVE})->Args({25, 5000, SPARSE_JUMPS_100})->Args({25, 5000, CONSECUTIVE_MODULO})-> \
+    Args({50, 1000, CONSECUTIVE})->Args({50, 1000, SPARSE_JUMPS_100})->Args({50, 1000, CONSECUTIVE_MODULO})
```

Comment:
Consider using `ArgsProduct` instead of manually expanding the matrix
```suggestion
    ArgsProduct({
      {10, 25, 50},
      {1000, 5000},
      {CONSECUTIVE, SPARSE_JUMPS_100, CONSECUTIVE_MODULO}
    })
```
Also, consider naming the arguments (with `ArgNames`)